File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<string>>mpp;
        for(string s:strs){
            string sorted=s;
            sort(sorted.begin(),sorted.end());
            mpp[sorted].push_back(s);
        }
        vector<vector<string>>ans;
        for(auto&it:mpp){
            ans.push_back(it.second);
        }
        return ans;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String,List<String>>mpp=new HashMap<>();
        for(String str:strs){
            char[]arr=str.toCharArray();
            Arrays.sort(arr);
            String sorted=new String(arr);
            mpp.putIfAbsent(sorted,new ArrayList<>());
            mpp.get(sorted).add(str);
        }
        List<List<String>>ans=new ArrayList<>();
        for(List<String>list:mpp.values()){
            ans.add(list);
        }
        return ans;
    }
}